### PR TITLE
fix some PVS warnings

### PR DESCRIPTION
**Pull request by justinmk**

### CODE REVIEW

File: src/nvim/tui/tui.c [hunk 1]
```diff
@@ -1857,7 +1857,7 @@ static void augment_terminfo(TUIData *data, const char *term,
               ut, "ext.set_underline_color", "\x1b[58:2:%p1%d:%p2%d:%p3%dm");
       } else {
           data->unibi_ext.set_underline_color = (int)unibi_add_ext_str(
-              ut, "ext.set_underline_color", "\x1b[58:2:%p1%d:%p2%d:%p3%dm");
+              ut, "ext.set_underline_color", "\x1b[58;2;%p1%d;%p2%d;%p3%dm");
```
Comment (justinmk) [inline src/nvim/tui/tui.c]: @blueyed I'm a bit surprised that this worked before. The terminal(s) you use support colon parameters?

Comment (blueyed) [reply src/nvim/tui/tui.c]: @justinmk 
Good catch. Yes, kitty appears to use colon parameters then.

Comment (blueyed) [reply src/nvim/tui/tui.c]: @justinmk 
Good catch, thanks. Yes, kitty appears to use colon parameters then.

Comment (blueyed) [reply src/nvim/tui/tui.c]: @justinmk 
Good catch, thanks. Yes, kitty appears to use colon parameters then.

**Pull request merged**